### PR TITLE
Add list of files to report of 'wp doctor check cache-flush' command

### LINKED ISSUE #184: wp_cache_flush warning without explanation. Happens with checksums verifier also
When using the check to detect usage of `wp_cache_fush()` function the output message should be more specific, because right now there is no indications about where it was detected (directory, file, etc).

It happens the same with the checksum validator. It throws a warning but doesn't display the file or files which doesn't match.

### PR DESCRIPTION
**Pull request by kiranpotphode**
Fixes #184

This PR adds enhancement for the `wp doctor check cache-flush` command as mentioned in issue #184

The output will show the list of files where `wp_cache_flush()` is used.

### CODE REVIEW

File: inc/checks/class-cache-flush.php [hunk 1]
```diff
@@ -36,6 +32,6 @@ public function run() {
 		}
 
 		$this->set_status( 'warning' );
-		$this->set_message( 'Use of wp_cache_flush() detected.' );
+		$this->set_message( 'Use of wp_cache_flush() detected in ' . implode( ', ', $this->_matches ) );
```
Comment (wojtekn) [inline inc/checks/class-cache-flush.php:35]: What do you think about improving the output readability, e.g., by removing the site path from the matches to display relative paths?

Comment (kiranpotphode) [reply inc/checks/class-cache-flush.php]: I have now modified the code to include relative paths only.

**Comment by wojtekn** [comment]
@kiranpotphode, thanks for adjusting the output format, it's now more readable.

I've just checked tests and found out we will need to update those as they rely on the old message format:

```
composer behat -- features/check-cache-flush.feature
```

**Comment by kiranpotphode** [comment]
> @kiranpotphode , thanks for adjusting the output format, it's now more readable.
> 
> I've just checked tests and found out we will need to update those as they rely on the old message format:
> 
> ```
> composer behat -- features/check-cache-flush.feature
> ```

@wojtekn  Thanks for pointing that out. 
I have updated the test now. I also updated the doc block comment for the class.

**Review by wojtekn — APPROVED** [review]
Looks good, thanks for updating the test!

**Comment by ernilambar** [comment]
@kiranpotphode Can you please rebase the PR with main branch? Now `behat.yml` file has been added. So GitHub actions will run with all functional tests.

**Comment by kiranpotphode** [comment]
> @kiranpotphode (https://profiles.wordpress.org/kiranpotphode) Can you please rebase the PR with main branch? Now `behat.yml` file has been added. So GitHub actions will run with all functional tests.

@ernilambar 
I have updated my branch with the latest code from the main branch.
However, I noticed that some tests are failing on GitHub actions.

But my tests on local are passing for the following command.
`composer behat -- features/check-cache-flush.feature`

Can you guide what might the problem?

**Comment by ernilambar** [comment]
Yah, I noticed. `wp_cache_flush()` is there in `sqlite-database-integration` which is used for Sqlite implementation. Probably you have tested with `mysql` in your setup. That is why you are not seeing error in local.

**Comment by kiranpotphode** [comment]
I have updated the tests. Now `wp_cache_flush()` related issues are fixed.

Still noticing some other failed tests for `WP trunk on PHP 8.4 ` Scenario though.

**Comment by ernilambar** [comment]
@kiranpotphode WP is still not compatible to PHP `8.4`. So we can ignore that now.

**Pull request merged**